### PR TITLE
Ladder5 v2

### PR DESCRIPTION
**Pull request by dirkgr**
This is the fifth version of the ladder. Maybe this one will stick.

Features:
 * Stability monitor
 * Based on Gemma, with a dash of Qwen
 * Minor fixes to `gemma_like()`, init, training on single GPUs (or MacBooks)

Also contains an untested init method that @YashasSamaga and I have been working on. It's not part of the ladder. Sorry for not pulling them apart.

---

> [!NOTE]
> **Medium Risk**
> Touches core weight initialization and attention/feed-forward init paths, which can materially change training behavior and checkpoint compatibility. Also adds new training/CLI pathways (single-rank mode and a large new ladder script) that need validation across CPU/CUDA and distributed setups.
> 
> **Overview**
> Introduces a new Gemma/Qwen-inspired ladder training script (`gemma_like_ladder.py`) with computed hyperparameter schedules, optional GatedDeltaNet usage, and a `--train-single` mode to force single-rank debugging by disabling DP/TP and selecting an appropriate distributed backend.
> 
> Adds `StabilityMonitorCallback`, which detects spikes in loss and grad norm and logs rolling/cumulative spike-rate metrics with checkpoint-resumable state, and wires it into callback exports and a new grad-norm metric constant.
> 
> Extends model/config initialization: adds `InitMethod.fan_in` (per-layer `1/√d_in` std, with embedding scaling support), applies it across attention/fused-attention and feed-forward (including MoE), fixes incorrect `self == InitMethod.*` checks to use `init_method`, and makes `TransformerConfig.gemma3_like()` configurable for attention `gate` and FFN `activation`. Includes new unit tests for `fan_in` init and the stability monitor.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3c7128a88144ad4e44b2244c0250a6676cb35911. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by dirkgr** [comment]
All remaining test failures are due to missing secrets, which is fine.

File: src/olmo_core/nn/attention/recurrent.py [hunk 1]
```diff
@@ -269,11 +269,11 @@ def init_weights(
...
         self.dt_bias.copy_(inv_dt)
 
-        if self == InitMethod.llama:
+        if init_method == InitMethod.llama:
             std = std / (2 * num_blocks) ** 0.5
-        elif self == InitMethod.llama_depth:
+        elif init_method == InitMethod.llama_depth:
             std = std / (2 * (block_idx + 1)) ** 0.5
-        elif self == InitMethod.normalized:
+        elif init_method == InitMethod.normalized:
```
Comment (dirkgr) [inline src/olmo_core/nn/attention/recurrent.py:281]: Fix of an unrelated issue

**Comment by dirkgr** [comment]
I think this is truly it.
